openedx/core/djangoapps/programs/utils.py

```diff
@@ -49,49 +46,25 @@ def get_programs(user, program_id=None):
         list of dict, representing programs returned by the Programs service.
         dict, if a specific program is requested.
     """
-    programs_config = ProgramsApiConfig.current()
-
-    # Bypass caching for staff users, who may be creating Programs and want
-    # to see them displayed immediately.
-    cache_key = programs_config.CACHE_KEY if programs_config.is_cache_enabled and not user.is_staff else None
-
-    programs = get_edx_api_data(programs_config, user, 'programs', resource_id=program_id, cache_key=cache_key)
-
-    # Mix in munged MicroMasters data from the catalog.
-    if not program_id:
-        programs += [
-            munge_catalog_program(micromaster) for micromaster in get_catalog_programs(user, type='MicroMasters')
-        ]
 
-    return programs
+    if use_catalog:
+        programs = [munge_catalog_program(program) for program in get_catalog_programs(user)]
```

Comment:
@MatthewPiatetsky I just had a startling realization about this with @clintonb. This is going to break when trying to paginate the catalog response. If you follow this through to `get_edx_api_data()`, you'll see that the method assumes `PageNumberPagination` (`page=n`). However, the catalog uses `LimitOffsetPagination` (`limit=20&offset=20`).

We've gotten away with this for pulling in MicroMasters because all MicroMasters in the catalog fit in a single response (there's no need to paginate the response). However, we have more XSeries programs than fit in a single response, so traversing the paginated response will be required. 

We could fix this by switching the catalog to `PageNumberPagination` (@clintonb, thoughts?), or by updating the LMS code to deal with `LimitOffsetPagination`. The latter is probably going to be easier, since it doesn't require updating any existing clients. This update could be made in a separate PR or as part of this one.